src/vcpkg/base/downloads.cpp

```diff
@@ -618,6 +617,61 @@ namespace vcpkg
         return res;
     }
 
+    ExpectedL<Unit> patch_file(const Filesystem& fs,
+                               StringView url,
+                               View<std::string> headers,
+                               const Path& file,
+                               std::size_t file_size,
+                               std::size_t chunk_size)
+    {
+        static constexpr StringLiteral guid_marker = "9a1db05f-a65d-419b-aa72-037fb4d0672e";
+
+        Command base_cmd;
+        base_cmd.string_arg("curl").string_arg("-X").string_arg("PATCH").string_arg("-w").string_arg(
+            "\\n" + guid_marker.to_string() + "%{http_code}\n");
+        for (auto&& header : headers)
+        {
+            base_cmd.string_arg("-H").string_arg(header);
+        }
+        base_cmd.string_arg(url);
+
+        auto file_ptr = fs.open_for_read(file, VCPKG_LINE_INFO);
+        std::vector<char> buffer(chunk_size);
```

Comment:
I see that the default size is 450 MB. And no limits.
Is there an alternative to reading it into a buffer first just to forward it to anothers command stdin?
(Remembering all those Raspi people which struggle to build `vcpkg` due to low memory...)

Reply:
The original way was to split the file on disk, but that's pretty hacky I think.

But I can decrease the buffer size. I'm not sure what you mean with limit. 

Are people really running a GitHub Runner server on a Raspi lmao 😄

Reply:
> Are people really running a GitHub Runner server on a Raspi lmao

Well, this is only the tool uploading the artifacts. Caching large artifacts is more important when build machine power is low.

Reply:
Ah okay. What buffer size do you think I should use? I can't make it really small or else the upload will be way slower than it would otherwise be.

Reply:
I don't know. I see the trade-offs and barriers.
- Can't make curl read chunks directly from (within) a large file.
- Can't feed (the vcpkg function running) curl with piecewise input. (IO buffer smaller than network chunks.)

Changing curl (tool) is out of scope here.
If the interface remains *running* curl instead of *calling* into libcurl, then it would be best to fix the second point.
If this is too intrusive, it might be helpful to have a way for the user to change the buffer size, or at least to turn of the buffering in case of trouble.